components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -48,6 +51,64 @@ constexpr char kWireguardConfigTemplate[] = R"(
   Endpoint = {vpn_server_hostname}:51821
 )";
 
+bool IsValidConfigDNS(const std::string& config_str) {
+  int match_count = 0, any_match_count = 0;
+
+  // Match our expected `DNS` field.
+  absl::string_view input(config_str);
+  while (match_count < 2 &&
+         re2::RE2::FindAndConsume(&input, R"((?m)^\s*DNS = 1\.1\.1\.1$)")) {
+    match_count++;
+  }
+
+  // Match any `DNS` field.
+  absl::string_view input_any(config_str);
+  while (any_match_count < 2 &&
+         re2::RE2::FindAndConsume(&input_any, R"((?m)^\s*DNS\s*=\s*.*$)")) {
```

Comment:
maybe should make this a case insensitive match unless we are sure wireguard configs are case sensitive?

Reply:
I left the expected match as case sensitive - because we control the exact formatting and it should only ever be `DNS` and `Endpoint`. However, I updated the query for "any" matches so that it would ID invalid ones. This also had the same problem as the whitespace - where:
```
DNS = 1.1.1.1
dns = 8.8.8.8
```
was allowed and failed verification. Added test cases for this too 😄